vault/identity_store_aliases.go

```diff
@@ -416,15 +421,46 @@ func (i *IdentityStore) pathAliasIDList() framework.OperationFunc {
 		}
 
 		var aliasIDs []string
+		aliasInfo := map[string]interface{}{}
+
+		type mountInfo struct {
+			MountType string
+			MountPath string
+		}
+		mountAccessorMap := map[string]mountInfo{}
+
 		for {
 			raw := iter.Next()
 			if raw == nil {
 				break
 			}
-			aliasIDs = append(aliasIDs, raw.(*identity.Alias).ID)
+			alias := raw.(*identity.Alias)
+			aliasIDs = append(aliasIDs, alias.ID)
+			aliasInfoEntry := map[string]interface{}{
+				"name":           alias.Name,
+				"canonical_id":   alias.CanonicalID,
+				"mount_accessor": alias.MountAccessor,
+			}
+
+			mi, ok := mountAccessorMap[alias.MountAccessor]
+			if ok {
+				aliasInfoEntry["mount_type"] = mi.MountType
+				aliasInfoEntry["mount_path"] = mi.MountPath
+			} else {
+				mi = mountInfo{}
+				if mountValidationResp := i.core.router.validateMountByAccessor(alias.MountAccessor); mountValidationResp != nil {
+					mi.MountType = mountValidationResp.MountType
+					mi.MountPath = mountValidationResp.MountPath
+					aliasInfoEntry["mount_type"] = mi.MountType
+					aliasInfoEntry["mount_path"] = mi.MountPath
+				}
+				mountAccessorMap[alias.MountAccessor] = mi
```

Comment:
Should this be inside the `if` condition above?

Reply:
No, because if `ok`, it's already in the map